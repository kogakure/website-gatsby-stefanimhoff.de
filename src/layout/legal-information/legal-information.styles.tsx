import styled from 'styled-components';
import { Link } from 'gatsby';

const LegalInformation = styled.div`
  align-items: center;
  display: flex;
  padding-left: clamp(15px, 2.4vw, 50px);
`;

const Copyright = styled.span`
  font-size: 14px;
  position: relative;
  top: 1px;
`;

const Author = styled.strong`
  font-size: 12px;
  letter-spacing: 0.05em;
  text-transform: uppercase;
`;

const AuthorLink = styled(Link)`
  color: var(--colorForeground);
  text-decoration: none;

  &:hover,
  &:focus {
    text-decoration: underline;
    text-decoration-color: var(--colorAccent) !important;
    text-decoration-thickness: 0.15em;
    text-underline-offset: 0.15em;
  }
`;

const Bullet = styled.span`
  font-size: 12px;
`;

const Year = styled.span`
  font-size: 14px;
`;

export const Styled = {
  Author,
  AuthorLink,
  Bullet,
  Copyright,
  LegalInformation,
  Year,
};
