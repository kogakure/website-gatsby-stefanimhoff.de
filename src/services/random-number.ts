export const getRandomNumber = (max: number) =>
  String(Math.floor(Math.random() * Math.floor(max)));
