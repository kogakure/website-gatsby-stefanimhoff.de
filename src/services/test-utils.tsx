import React from 'react';
import { render, RenderOptions, RenderResult } from '@testing-library/react';

import { EmojifyProvider, LocalizedDateProvider } from '../contexts';

const customRender = (
  node: React.ReactElement,
  options?: RenderOptions
): RenderResult => {
  const rendered = render(
    <EmojifyProvider>
      <LocalizedDateProvider>{node}</LocalizedDateProvider>
    </EmojifyProvider>,
    options
  );

  return {
    ...rendered,
    rerender: (ui: React.ReactElement, rerenderOptions?: RenderOptions) =>
      customRender(ui, { container: rendered.container, ...options }),
  };
};

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render };
