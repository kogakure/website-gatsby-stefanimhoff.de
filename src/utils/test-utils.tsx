import React from 'react';
import { render, RenderOptions, RenderResult } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';

import { EmojifyProvider, LocalizedDateProvider } from '../contexts';
import { ThemeType } from '../typings/Theme';
import { lightTheme } from '../themes';

interface CustomTheme {
  theme: ThemeType;
}

type RenderOptionsWithTheme = RenderOptions & CustomTheme;

const customRender = (
  node: React.ReactElement,
  options: RenderOptionsWithTheme = { theme: lightTheme }
): RenderResult => {
  const { theme } = options;
  const rendered = render(
    <ThemeProvider theme={theme}>
      <EmojifyProvider>
        <LocalizedDateProvider>{node}</LocalizedDateProvider>
      </EmojifyProvider>
    </ThemeProvider>,
    options
  );

  return {
    ...rendered,
    rerender: (
      ui: React.ReactElement,
      rerenderOptions?: RenderOptionsWithTheme
    ) => customRender(ui, { container: rendered.container, ...options }),
  };
};

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render };
