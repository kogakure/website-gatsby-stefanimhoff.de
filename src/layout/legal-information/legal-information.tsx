import * as React from 'react';

import { Styled } from './legal-information.styles';

export const LegalInformation: React.FC = ({ ...props }) => (
  <Styled.LegalInformation {...props}>
    <div>
      <Styled.Copyright>© </Styled.Copyright>
      <Styled.Author>
        <Styled.AuthorLink to="/imprint/">Stefan Imhoff</Styled.AuthorLink>
      </Styled.Author>
      <Styled.Bullet> / </Styled.Bullet>
      <Styled.Year>
        <time dateTime={String(new Date().getFullYear())}>
          2006–{new Date().getFullYear()}
        </time>
      </Styled.Year>
    </div>
  </Styled.LegalInformation>
);
