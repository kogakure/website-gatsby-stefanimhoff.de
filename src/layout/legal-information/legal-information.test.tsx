import React from 'react';
import { axe } from 'jest-axe';

import { render } from '../../services/test-utils';

import { LegalInformation } from './legal-information';

describe('LegalInformation', () => {
  test('renders correctly', () => {
    const { container } = render(<LegalInformation />);
    const mockDate = new Date(1590275951821);
    const spy = jest
      .spyOn(global, 'Date')
      .mockImplementation(() => String(mockDate));

    expect(container.firstChild).toMatchSnapshot();

    spy.mockRestore();
  });

  test('is accessible', async () => {
    const { container } = render(<LegalInformation />);
    const results = await axe(container);

    expect(results).toHaveNoViolations();
  });
});
