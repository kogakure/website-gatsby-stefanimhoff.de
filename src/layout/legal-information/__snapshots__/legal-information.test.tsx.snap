// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LegalInformation renders correctly 1`] = `
.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding-left: clamp(15px,2.4vw,50px);
}

.c1 {
  font-size: 14px;
  position: relative;
  top: 1px;
}

.c2 {
  font-size: 12px;
  -webkit-letter-spacing: 0.05em;
  -moz-letter-spacing: 0.05em;
  -ms-letter-spacing: 0.05em;
  letter-spacing: 0.05em;
  text-transform: uppercase;
}

.c3 {
  color: var(--colorForeground);
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c3:hover,
.c3:focus {
  -webkit-text-decoration: underline;
  text-decoration: underline;
  -webkit-text-decoration-color: var(--colorAccent) !important;
  text-decoration-color: var(--colorAccent) !important;
  -webkit-text-decoration-thickness: 0.15em;
  text-decoration-thickness: 0.15em;
  text-underline-offset: 0.15em;
}

.c4 {
  font-size: 12px;
}

.c5 {
  font-size: 14px;
}

<div
  class="c0"
>
  <div>
    <span
      class="c1"
    >
      © 
    </span>
    <strong
      class="c2"
    >
      <a
        class="c3"
        href="/imprint/"
      >
        Stefan Imhoff
      </a>
    </strong>
    <span
      class="c4"
    >
       / 
    </span>
    <span
      class="c5"
    >
      <time
        datetime="2020"
      >
        2006–
        2020
      </time>
    </span>
  </div>
</div>
`;
